### DISCUSSION TITLE
Sembunyikan/Hapus Iklan Produk Promoted pada Hasil Pencarian di Marketplace (Tokopedia & Shopee)

### DISCUSSION THREAD

**Discussion by sandycorzeta** [realodix/AdBlockID#1092] [Personal Filter]
Halo, saya ingin mencoba untuk berkontribusi menambahkan filter annoyance untuk web marketplace yg menurut saya cukup redundant dalam menampilkan item produk. Setiap melakukan pencarian, pada marketplace manapun biasanya selalu ditambahkan iklan produk promoted di bagian baris pertama dari hasil pencarian. Seperti pada gambar tangkapan layar yang saya lampirkan, terdapat redundansi dari item produk yang ditampilkan dan terkadang item produk dengan tanda "iklan" tersebut mendapat perlakuan pengecualian dari filter custom pencarian yang mana membuat user experience menjadi membingungkan.

![image](https://user-images.githubusercontent.com/4186454/147617923-ad1f73b8-0877-4272-acd4-f82261e57c99.png)

Oleh karena itu, saya mencoba untuk berbagi personal filter yang sekiranya dapat menangani hal semacam ini. Berikut list personal filter yang saya gunakan untuk Tokopedia & Shopee

```
## Tokopedia 
! Untuk di halaman dengan url parameter /search. contoh : https://www.tokopedia.com/search?st=product&q=Ayam
tokopedia.com##:xpath(//div[@data-testid="divSRPContentProducts"]//div//div[.//div//div//div//div//div//div//div//span[@data-testid="divSRPTopadsIcon"]]):remove()

! Untuk di halaman dengan url parameter /find. contoh : https://www.tokopedia.com/find/macbook-air
tokopedia.com##:xpath(//div[@data-testid="cntrFindProductsResult"]//div//div[.//div//div//div//div//div//div//div//span[@data-testid="lblTopads"]]):remove()

! Untuk di halaman kategori dengan url parameter /p/* , contoh : https://www.tokopedia.com/p/fashion-pria/tas-pria/tas-selempang-pria
tokopedia.com##:xpath(//div[@data-testid="lstCL3ProductList"]//div[.//a[@label-atm="taProduct"]]):remove()
tokopedia.com##:xpath(//div[@data-testid="catBreadCrumbWrapper"]//div//div//div[.//div//div//div[@data-impressed="true"]]):remove()

## Shopee
shopee.co.id##:xpath(//div[@data-sqe="item"][.//a//div//div//div//div[@data-sqe="ad"]]):remove()

```

Saya menggunakan uBlock sebagai pemblokir iklan dan menggunakan syntax :remove() untuk menghapus elemen item tersebut daripada menyembunyikannya. Karena apabila disembuyikan lazy-loading akan membuat iklan produk promoted tersebut muncul kembali.

**Comment by realodix** [comment]
Halo, terima kasih sudah mempertimbangkan untuk berkontribusi dan berbagi personal filter.

Saya sudah merevisi filter untuk tokped dan shopee untuk menggunakan `:remove()` sesuai saran dari Anda.
- tokopedia.com (https://github.com/realodix/AdBlockID/commit/82c06aafb2446f20316d0e8f54e97e5d4d334d52)
- shopee.co.id (https://github.com/realodix/AdBlockID/commit/3dc89b68768cc5910d38ae0717842087f02f91a2)

Tetapi Saya tidak menggunakan `:xpath()`, karena Saya tidak tahu apa keunggulan `:xpath()` dan dalam kondisi seperti apa harus menggunakan `:xpath()`. Sebagai gantinya Saya menggunakan `:matches-path()` dan `:has()`.

**Reply by sandycorzeta** [reply]
Terima kasih banyak, saya belum mengetahui mengenai keuntungan untuk menggunakan `:xpath()` karena sewaktu saya mencari tutorial untuk ublock yang dapat melakukan prosedural, tutorial tersebut menggunakan `:xpath()` dan belum tau sepenuhnya bahwa `:matches-path()` dan `:has()` dapat menggantikan itu dengan syntax yg lebih sederhana.
